src/routes/setup/+page.svelte

```diff
@@ -1,13 +1,107 @@
-<script>
-	import { PlayerFieldSet, Well } from '$lib/components';
+<script lang="ts">
+	import Rules from '$lib/components/RuleForm.svelte';
+	import Players from '$lib/components/PlayerForm.svelte';
 	import logo from '$lib/assets/brand.svg';
+	import { goto, onNavigate } from '$app/navigation';
+	import { NineBallGame, Player } from '$lib';
+	import Toast from "$lib/components/Toast.svelte"
+
+	export let data;
+	let { game } = data
+
+	let isRedirect = false;
+	let redirectInt: NodeJS.Timeout
+	
+	onNavigate((nav) => {
+		if (nav.from?.route.id === nav.to?.route.id) {
+			isRedirect = true

```

Comment:
This is a great start, but I think we can create a store in the root `+layout.ts` 
```typescript
const toast = writable<ComponentProps<Toast> | null>(null)
```
Then instead of looking for redirects we can just show a toast if the store is non-null